src/main/java/no/unit/nva/handlers/HandlerWithEventualConsistency.java

```diff
@@ -32,16 +28,16 @@ protected HandlerWithEventualConsistency(Class<I> iclass, Environment environmen
         return eventuallyConsistentObject;
     }
 
-    private void waitForEventualConsistency() throws UnexpectedException {
+    private void waitForEventualConsistency() {
         try {
             Thread.sleep(WAITING_TIME);
         } catch (InterruptedException e) {
             logger.error(INTERRUPTION_ERROR, e);
-            throw new UnexpectedException(INTERRUPTION_ERROR, e);
+            throw new RuntimeException(INTERRUPTION_ERROR, e);
```

Comment:
Since it is unexpected the exception it does not make much sense to try to catch an Unexpected exception. Better to throw a Runtime exception that will be caught and reported by our framework